### DISCUSSION TITLE
How to get metric for Total number of active connections

### DISCUSSION THREAD

**Discussion by arjaiswa** [nginx/kubernetes-ingress#1870] [General]
I want to get the metric 
1. Total Active connections
2. Total no of connections

How I can get these metrics ?

I can see the stubs : But this is not correct because there is no active connections correctly.
Active connections: 2 
server accepts handled requests
 27472 27472 9673 
Reading: 0 Writing: 1 Waiting: 1

**Comment by pleshakov** [comment]
Hi @arjaiswa 

As part of a configuration reload, the IC process reloads NGINX (via `nginx -s reload`) and then sends an HTTP request to check that the new configuration was applied. This check will establish a keep alive connection to NGINX, which will be reflected in the `Active connections`.

When you view the stub status, it is also counted as an active connection. 

Total number of handled connections correspond to the handled metric.

Note that you can also get those metrics in Prometheus format via `:9113/metrics` endpoint. For example:
```
# HELP nginx_ingress_nginx_connections_accepted Accepted client connections
# TYPE nginx_ingress_nginx_connections_accepted counter
nginx_ingress_nginx_connections_accepted{class="nginx"} 17
# HELP nginx_ingress_nginx_connections_active Active client connections
# TYPE nginx_ingress_nginx_connections_active gauge
nginx_ingress_nginx_connections_active{class="nginx"} 3
# HELP nginx_ingress_nginx_connections_handled Handled client connections
# TYPE nginx_ingress_nginx_connections_handled counter
nginx_ingress_nginx_connections_handled{class="nginx"} 17
# HELP nginx_ingress_nginx_connections_reading Connections where NGINX is reading the request header
# TYPE nginx_ingress_nginx_connections_reading gauge
nginx_ingress_nginx_connections_reading{class="nginx"} 0
# HELP nginx_ingress_nginx_connections_waiting Idle client connections
# TYPE nginx_ingress_nginx_connections_waiting gauge
nginx_ingress_nginx_connections_waiting{class="nginx"} 2
# HELP nginx_ingress_nginx_connections_writing Connections where NGINX is writing the response back to the client
# TYPE nginx_ingress_nginx_connections_writing gauge
nginx_ingress_nginx_connections_writing{class="nginx"} 1
# HELP nginx_ingress_nginx_http_requests_total Total http requests
# TYPE nginx_ingress_nginx_http_requests_total counter
nginx_ingress_nginx_http_requests_total{class="nginx"} 86
```

Hope this clarifies

**Comment by arjaiswa** [comment]
Yes its good .
I want to know the reliability of :9113/metric or are there any way to read Prometheus metric from local filesystem.

**Reply by brianehlert** [reply]
The Prometheus endpoint is continuously reporting the current situation directly from NGINX.
Using NGINX Plus yields more detail than NGINX OSS.
The Prometheus endpoint runs as a pod and would be restarted by K8s if anything happened.
This detail is not written to the local file system, but rather interpreted directly from NGINX.

**Comment by jjngx** [comment]
Answered 4 years ago. No follow up. Closing.
